gdc_client/parcel/segment.py

```diff
@@ -120,83 +122,118 @@ def validate_segment_md5sums(self, path=None):
-                '{1}'. Either this file was downloaded to a different
-                location, the state file was moved, or the state file
-                was deleted.  Parcel refuses to claim the file has
-                been successfully downloaded and will restart the
-                download.\n"""
-                ).format(
-                    (
-                        self.download.path
-                        if os.path.isfile(self.download.path)
-                        else self.download.temp_path
-                    ),
-                    self.download.state_path,
+            log.warning("Redownloading {0} corrupt segments.".format(corrupt_segments))
+
+    def recover_intervals(self):
+        """Recreate list of completed intervals and calculate remaining work pool"""
+        state_file_exists = os.path.isfile(self.download.state_path)
+        download_file_exists = os.path.isfile(self.download.path)
+        temporary_file_exists = os.path.isfile(self.download.temp_path)
+
+        # If the state file does not exist, treat as first time download
+        if not state_file_exists:
+            log.debug(
+                "State file {0} does not exist. Beginning new download...".format(
+                    self.download.state_path
                 )
             )
-            return
-
-        if not os.path.isfile(self.download.state_path):
-            self.download.setup_file()
-            return
+            return False
 
-        # If there is a file at load_path, attempt to remove
-        # downloaded sections from work_pool
         log.debug(
             "Found state file {0}, attempting to resume download".format(
                 self.download.state_path
             )
         )
-
-        if not os.path.isfile(self.download.path) and not os.path.isfile(
-            self.download.temp_path
-        ):
-            log.warning(
-                STRIP(
-                    """State file found at '{0}' but no file for {1}.
-                Restarting entire download.""".format(
-                        self.download.state_path, self.download.url
-                    )
-                )
-            )
-            return
+        # Attempt to load completed segments from state file
         try:
             with open(self.download.state_path, "rb") as f:
                 self.completed = pickle.load(f)
             assert isinstance(
                 self.completed, IntervalTree
             ), "Bad save state: {0}".format(self.download.state_path)
         except Exception as e:
-            self.completed = IntervalTree()
-            log.error("Unable to resume file state: {0}".format(str(e)))
-        else:
-            self.validate_segment_md5sums(
-                (
+            # An error has occured while loading state file.
+            # Treat as entire file download and recreate temporary file
+            log.error(
```

Comment:
When we log/catch exceptions its really important to capture the type and the stack trace for the exception. You are eating most of the information which would actually be useful in debugging this code if an unexpected exception was happening in the above block.